keras/utils/layer_utils.py

```diff
@@ -24,12 +24,29 @@ def print_summary(model, line_length=None, positions=None, print_fn=print):
         sequential_like = True
     else:
         sequential_like = True
-        for v in model.nodes_by_depth.values():
+        nodes_by_depth = model.nodes_by_depth.values()
+        nodes = []
+        for v in nodes_by_depth:
             if (len(v) > 1) or (len(v) == 1 and len(v[0].inbound_layers) > 1):
                 # if the model has multiple nodes or if the nodes have multiple inbound_layers
                 # the model is no longer sequential
                 sequential_like = False
                 break
+            nodes += v
+        if sequential_like:
+            # search for shared layers
```

Comment:
I don't get the criteria you use for shared layers. A comment here would help

Reply:
A layer is shared  in a model if it has 2 or more inbound nodes and those inbound nodes are part of the graph.